Add poetry to manage dependencies

naddeoa opened the pull request:
This change introduces poetry as a dependency management tool. Poetry will be replacing our requirements files, allow us to specify dev specific dependencies in a managed way, and generate lock files so we can reproduce installations on other machines.

naddeoa committed:
Got everything working except for one test

naddeoa committed:
added bump2version

naddeoa committed:
Fix broken test

This probably broke because of an update to Pillow but its unclear why.
The test may not be a great test since it asserts the number of
features. That number can apparently change across different versions.

naddeoa committed:
Use `poetry run` to run python commands in the Makefile

naddeoa committed:
Update makefile echo functions

I'l figure color out later

naddeoa committed:
Use poetry instead of pip dudring the github builds

naddeoa committed:
Add tox and use poetry for a few one off commands

naddeoa committed:
Update tox to not try and install dependencies

naddeoa committed:
added missing dependencies

naddeoa committed:
Fix linter

naddeoa committed:
added missing dependencies

naddeoa committed:
Attempting to fix tox

naddeoa committed:
Use makefile in the github actions

andyndang commented:
```
Running setup.py develop for whylogs
    ERROR: Command errored out with exit status 1:
     command: /Users/runner/work/whylogs/whylogs/.tox/py39/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/runner/work/whylogs/whylogs/setup.py'"'"'; __file__='"'"'/Users/runner/work/whylogs/whylogs/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: /Users/runner/work/whylogs/whylogs/
    Complete output (3 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    ModuleNotFoundError: No module named 'setuptools'
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/runner/work/whylogs/whylogs/.tox/py39/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/runner/work/whylogs/whylogs/setup.py'"'"'; __file__='"'"'/Users/runner/work/whylogs/whylogs/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.

```
CI is failing atm

naddeoa committed:
removed tox, working on using github for multi version testing

naddeoa committed:
saving

naddeoa committed:
saving

naddeoa committed:
saving

naddeoa committed:
updating the email address to whylabs

naddeoa committed:
Support python 3.6 as well

naddeoa committed:
Fix excel test for python 3.6

Pandas doesn't work with excel in our version unfortunately.

naddeoa committed:
Remove excel format support

We'll have to see about adding it back after upgrading pandas or using
something besides pandas to support it.

lalmei commented on tests/unit/io/test_file_loader.py:
```diff
@@ -19,9 +19,11 @@ def test_image_loader(image_files):
 
 @pytest.mark.skipif(sys.version_info < (3, 8), reason="requires python3.7 or higher")
 def test_file_loader(file_list):
+    filefmts = [(pd.DataFrame, "csv"), (list, "jsonl")]
```
I dont see the need to support these old versions of python or pandas

lalmei replied:
especially when it removes functionality

naddeoa replied:
I agree, my current plan is to get it working and checked in with the version that we have already (this bad one) and bring it up again to talk about dropping support for 3.6 for supporting excel in a different way than pandas.

naddeoa committed:
Remove excel files from test

andyndang reviewed (approved):
LGTM
